p2p/protocol/identify/id.go

```diff
@@ -105,13 +106,15 @@ func NewIDService(ctx context.Context, h host.Host, opts ...Option) *IDService {
 		userAgent = cfg.userAgent
 	}
 
+	hostCtx, cancel := context.WithCancel(context.Background())
```

Comment:
being able to have the ID service run within a larger context continues to seem valuable.

Reply:
I don't see why. It's a separate service and exposes a `Close` function we can call to shut it down. This is similar to the DHT<-> Routing Table relationship and the same idea works fine for us there.

What are your concerns here ?

Reply:
if a consumer is spinning up a bunch of separate services, it's nice to have them all wired up to a central context, and then cancel that context to shut things down, rather than explicitly calling close on each service.

Reply:
I see where you are coming from but we've had problems in the past using contexts for cancellation across service boundaries. Please take a look at https://github.com/libp2p/go-libp2p-kbucket/pull/50#discussion_r380351569 & the linked discussions in that comment.

The most important benefit to me is being able control the order in which the services, sub-components etc are shut down.